pkg/controller/podprobemarker/ppm_event_handler.go

```diff
@@ -0,0 +1,248 @@
+
+	appsalphav1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	"github.com/openkruise/kruise/pkg/util"
+	utilclient "github.com/openkruise/kruise/pkg/util/client"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ handler.EventHandler = &enqueueRequestForPodProbeMarker{}
+
+type enqueueRequestForPodProbeMarker struct{}
+
+func (p *enqueueRequestForPodProbeMarker) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	p.queue(q, evt.Object)
+}
+
+func (p *enqueueRequestForPodProbeMarker) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+}
+
+func (p *enqueueRequestForPodProbeMarker) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+}
+
+func (p *enqueueRequestForPodProbeMarker) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	p.queue(q, evt.ObjectNew)
+}
+
+func (p *enqueueRequestForPodProbeMarker) queue(q workqueue.RateLimitingInterface, obj runtime.Object) {
+	ppm, ok := obj.(*appsalphav1.PodProbeMarker)
+	if !ok {
+		return
+	}
+	q.Add(reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: ppm.Namespace,
+			Name:      fmt.Sprintf("%s#%s", ReconPodProbeMarker, ppm.Name),
+		},
+	})
+}
+
+var _ handler.EventHandler = &enqueueRequestForNodePodProbe{}
+
+type enqueueRequestForNodePodProbe struct{}
+
+func (p *enqueueRequestForNodePodProbe) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	obj, ok := evt.Object.(*appsalphav1.NodePodProbe)
+	if !ok {
+		return
+	}
+	p.queue(q, obj)
+}
+
+func (p *enqueueRequestForNodePodProbe) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+}
+
+func (p *enqueueRequestForNodePodProbe) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+}
+
+func (p *enqueueRequestForNodePodProbe) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	new, ok := evt.ObjectNew.(*appsalphav1.NodePodProbe)
+	if !ok {
+		return
+	}
+	old, ok := evt.ObjectOld.(*appsalphav1.NodePodProbe)
+	if !ok {
+		return
+	}
+	if reflect.DeepEqual(new.Status, old.Status) {
+		return
+	}
+	p.queue(q, new)
+}
+
+func (p *enqueueRequestForNodePodProbe) queue(q workqueue.RateLimitingInterface, npp *appsalphav1.NodePodProbe) {
+	q.Add(reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Namespace: npp.Namespace,
+			Name:      fmt.Sprintf("%s#%s", ReconNodePodProbe, npp.Name),
+		},
+	})
+}
+
+var _ handler.EventHandler = &enqueueRequestForPod{}
+
+type enqueueRequestForPod struct {
+	reader client.Reader
+}
+
+func (p *enqueueRequestForPod) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+}
```

Comment:
shall we enqueue pod missing in the nodepodprobe after kruise restarts

Reply:
PodProbeMarker also trigger reconcile, and will list all pod to create nodePodProbe.spec